tests/foreman/virtwho/api/test_esx.py

```diff
@@ -82,67 +83,19 @@ def test_positive_deploy_configure_by_id(
         :CaseImportance: High
         """
         assert virtwho_config.status == 'unknown'
-        command = get_configure_command(virtwho_config.id, default_org.name)
-        hypervisor_name, guest_name = deploy_configure_by_command(
-            command, form_data['hypervisor_type'], debug=True, org=default_org.label
-        )
-        virt_who_instance = (
-            target_sat.api.VirtWhoConfig()
-            .search(query={'search': f'name={virtwho_config.name}'})[0]
-            .status
-        )
-        assert virt_who_instance == 'ok'
-        hosts = [
-            (
-                hypervisor_name,
-                f'product_id={settings.virtwho.sku.vdc_physical} and type=NORMAL',
-            ),
-            (
-                guest_name,
-                f'product_id={settings.virtwho.sku.vdc_physical} and type=STACK_DERIVED',
-            ),
-        ]
-        for hostname, sku in hosts:
-            host = target_sat.cli.Host.list({'search': hostname})[0]
-            subscriptions = target_sat.cli.Subscription.list(
-                {'organization': default_org.name, 'search': sku}
+        if deploy_type == "id":
```

Comment:
I've noticed a significant amount of repetitive code for the Id and Script sections. To simplify this, I suggest creating a factory fixture that handles the configuration setup for you. By incorporating this parameterization into indirect parameterization, the code will be more streamlined and maintainable. If you need assistance with implementing this approach, I'd be happy to provide guidance.

 If you need to refer to some examples. here you go:     https://github.com/SatelliteQE/robottelo/blob/5b43b56d5bb3deb7ea4140398c4f061d6485246f/pytest_fixtures/component/satellite_auth.py#L174